### PR TITLE
API-1835: Use dynamic operator client

### PR DESCRIPTION
**Pull request by deads2k**
/assign @p0lyn0mial @liouk 

For future refactors we'll want to use the "standard" generic operator client.  This refactors as much as possible to use that.

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -633,7 +641,8 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
 			Optional: true,
 		}},
 		operatorCtx.kubeInformersForNamespaces.InformersFor("openshift-oauth-apiserver"),
-		revisionclient.New(operatorCtx.operatorClient, operatorCtx.operatorClient.Client),
+		// TODO looks like the concept of revisions has leaked into at least one non-static pod operator.  Probably end up making this a flavor of regular OperatorStatus?
+		revisionclient.New(operatorCtx.operatorClient, operatorClient),
```
Comment (deads2k) [inline pkg/operator/starter.go:645]: I think this writes `.status.LatestAvailableRevision`

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -534,7 +542,7 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
 	migrator := migrators.NewKubeStorageVersionMigrator(migrationClient, migrationInformer.Migration().V1alpha1(), operatorCtx.kubeClient.Discovery())
 
 	authAPIServerWorkload := workload.NewOAuthAPIServerWorkload(
-		operatorCtx.operatorClient.Client,
+		operatorClient,
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/operator/starter.go:545]: reads [.Status.OAuthAPIServer.LatestAvailableRevision](https://github.com/openshift/cluster-authentication-operator/blob/3ac53812cf416ee33f873497ba25bc13a6dcda28/pkg/operator/workload/sync_openshift_oauth_apiserver.go#L154) that's why we cannot use the generic client as it is today.

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -522,6 +525,11 @@ func prepareOauthAPIServerOperator(ctx context.Context, controllerContext *contr
 	}
 	apiregistrationInformers := apiregistrationinformers.NewSharedInformerFactory(apiregistrationv1Client, 10*time.Minute)
 
+	operatorClient, err := operatorconfigclient.NewForConfig(controllerContext.KubeConfig)
```
Comment (p0lyn0mial) [inline pkg/operator/starter.go:528]: renaming `operatorClient` to `typedOperatorClient` would make reading the code slightly more pleasant.

Comment (p0lyn0mial) [reply pkg/operator/starter.go:528]: I think that in line [124](https://github.com/openshift/cluster-authentication-operator/blob/3ac53812cf416ee33f873497ba25bc13a6dcda28/pkg/operator/starter.go#L124) we already created the typed client. Should we reuse it instead ?

File: pkg/operator/starter.go [hunk 4]
```diff
@@ -9,23 +9,7 @@ import (
...
-	certinformers "k8s.io/client-go/informers"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/klog/v2"
-	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
-	apiregistrationclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
-	apiregistrationinformers "k8s.io/kube-aggregator/pkg/client/informers/externalversions"
-	utilpointer "k8s.io/utils/pointer"
-	kubemigratorclient "sigs.k8s.io/kube-storage-version-migrator/pkg/clients/clientset"
-	migrationv1alpha1informer "sigs.k8s.io/kube-storage-version-migrator/pkg/clients/informer"
+	operatorconfigclient "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
```
Comment (p0lyn0mial) [inline pkg/operator/starter.go:13]: remove the empty line.

File: pkg/operator/starter.go (hunk 3 above)
Comment (deads2k) [reply pkg/operator/starter.go:528]: > I think that in line [124](https://github.com/openshift/cluster-authentication-operator/blob/3ac53812cf416ee33f873497ba25bc13a6dcda28/pkg/operator/starter.go#L124) we already created the typed client. Should we reuse it instead ?

Different function used to drive informers on a bad resync cycle.  One of the refactors will remove the one from line 124.

**Comment by deads2k** [comment]
Comments addressed

File: pkg/controllers/payload/payload_config_controller.go [hunk 5]
```diff
@@ -88,8 +90,8 @@ func NewPayloadConfigController(kubeInformersForTargetNamespace informers.Shared
 	).ResyncEvery(wait.Jitter(time.Minute, 1.0)).WithSync(c.sync).ToController("PayloadConfig", recorder.WithComponentSuffix("payload-config-controller"))
 }
 
-func (c *payloadConfigController) getAuthConfig(ctx context.Context) (*operatorv1.Authentication, []operatorv1.OperatorCondition) {
-	operatorConfig, err := c.auth.Authentications().Get(ctx, "cluster", metav1.GetOptions{})
+func (c *payloadConfigController) getAuthConfig(ctx context.Context) (*operatorv1.OperatorSpec, []operatorv1.OperatorCondition) {
```
Comment (liouk) [inline pkg/controllers/payload/payload_config_controller.go:93]: `ctx` is unused.

```suggestion
func (c *payloadConfigController) getAuthConfig() (*operatorv1.OperatorSpec, []operatorv1.OperatorCondition) {
```

**Review by liouk — CHANGES_REQUESTED** [review]

File: pkg/controllers/payload/payload_config_controller.go (hunk 5 above)
Comment (deads2k) [reply pkg/controllers/payload/payload_config_controller.go:93]: > `ctx` is unused.

Let's plumb it through everywhere for future usage.  It's a good practice.

**Comment by liouk** [comment]
/lgtm
/hold

holding for @p0lyn0mial to finish his review

**Comment by deads2k** [comment]
moving ahead, will take post merge comments with a nag via slack.  I think everything Lucasz commented on was directly handled.

/hold cancel

**Pull request merged**